src/UI/Palette/V2.elm

```diff
@@ -0,0 +1,653 @@
+import Element.Border as Border
+import Element.Font as Font
+import UI.Internal.Colors as Internal exposing (..)
+import UI.Utils.Element exposing (colorSetOpacity)
+
+
+{-| `Palette.Color` holds data about some desired color.
+-}
+type Color
+    = Color Properties Options
+
+
+type alias Properties =
+    { hue : Hue, shade : Shade }
+
+
+type alias Options =
+    { contrast : Bool
+    , alpha : Float
+    }
+
+
+{-| The design system describes four main entries that here are called Hues.
+
+A hue is one of the five pure colors of the palette.
+
+-}
+type Hue
+    = HueGray
+    | HueBlue
+    | HueGreen
+    | HueYellow
+    | HueRed
+
+
+{-| Each [hue](UI-Palette#Hue) can be paired with eight different shades.
+
+The shades are 800, 700, 600, 500, 400, 300, 200 and 100.
+
+-}
+type Shade
+    = Shade800
+    | Shade700
+    | Shade600
+    | Shade500
+    | Shade400
+    | Shade300
+    | Shade200
+    | Shade100
+
+
+{-| Given a hue and shade it constructs a color.
+
+    Palette.color hueBlue shade600
+
+-}
+color : Hue -> Shade -> Color
+color hue shade =
+    Color (Properties hue shade) defaultOptions
+
+
+{-| Shorthand for setting background colors
+
+    Element.row
+        [ Palette.blue600 |> Palette.toBackgroundColor
+        ]
+        []
+
+-}
+toBackgroundColor : Color -> Element.Attribute msg
+toBackgroundColor =
+    toElementColor >> Background.color
+
+
+{-| Shorthand for setting font colors
+
+    Element.row
+        [ Palette.blue600 |> Palette.toFontColor
+        , Font.size 16
+        , Font.justify
+        ]
+        []
+
+-}
+toFontColor : Color -> Element.Attribute msg
+toFontColor =
+    toElementColor >> Font.color
+
+
+{-| Shorthand for setting border colors
+
+    Element.row
+        [ Palette.blue600 |> Palette.toBorderColor
+        , Border.width 2
+        , Border.solid
+        ]
+        []
+
+-}
+toBorderColor : Color -> Element.Attribute msg
+toBorderColor =
+    toElementColor >> Border.color
+
+
+{-| Manually transforms a [`Palette.Color`](UI-Palette#Color) into an Elm-UI-compatible color.
+
+    let
+        backgroundColor =
+            Palette.color tonePrimary brightnessMiddle
+    in
+    Element.el
+        [ backgroundColor
+            |> Palette.setContrasting True
+            |> Palette.toElementColor
+            |> Element.Font.color
+        , backgroundColor
+            |> Palette.toElementColor
+            |> Element.Background.color
+        ]
+    <|
+        Element.text "Hello World!"
+
+-}
+toElementColor : Color -> Element.Color
+toElementColor (Color { hue, shade } { alpha, contrast }) =
+    toColor contrast shade hue
+        |> colorSetOpacity alpha
+
+
+{-| Inverts a color for contrast. Useful for contrasting text with the background.
+
+    backgroundColor
+        |> Palette.setContrasting True
+        |> Palette.toElementColor
+        |> Element.Font.color
+
+-}
+setContrasting : Bool -> Color -> Color
+setContrasting enabled (Color prop opt) =
```

Comment:
I would rather drop this. I don't recall any use-case for it. It is leftover from V1.